File: origincodes/iOS 底层实现Demos/iOS 底层实现Demos/ViewController.m
//
//  ViewController.m
//  iOS 底层实现Demos
//
//  Created by 冯才凡 on 2019/11/21.
//  Copyright © 2019 冯才凡. All rights reserved.
//

#import "ViewController.h"

#import "SubController2.h"

#import "FCFNotificationCerter.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
//    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(notificate:) name:@"test" object:nil];
    
    
    [[FCFNotificationCerter defaultCenter] addObserver:self selector:@selector(notificate:) name:@"test" object:nil];
    
    
}

- (IBAction)btn1Action:(id)sender {
    SubController2 * vc =  [[SubController2 alloc] init];
    vc.view.frame = CGRectMake(0, 0, UIScreen.mainScreen.bounds.size.width, UIScreen.mainScreen.bounds.size.height);
    vc.view.backgroundColor = [UIColor lightGrayColor];
   
    [self presentViewController:vc animated:YES completion:nil];
    
}

- (IBAction)btn2Action:(id)sender {
    
}

- (void)notificate:(NSNotification *)noti {
    NSLog(@"receive notification1");
}



@end

File: WikiDemo/AlamofireDemo/AlamofireDemo/ViewController.swift
//
//  ViewController.swift
//  AlamofireDemo
//
//  Created by 冯才凡 on 2020/7/30.
//  Copyright © 2020 冯才凡. All rights reserved.
//

import UIKit
import Alamofire

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let alamofirebtn = UIButton(frame: CGRect(x: 100, y: 100, width: 100, height: 44))
        alamofirebtn.setTitle("alamofire", for: .normal)
        alamofirebtn.setTitleColor(UIColor.white, for: .normal)
        alamofirebtn.addTarget(self, action: #selector(alamofirebtnAction), for: .touchUpInside)
        view.addSubview(alamofirebtn)
        
    }


}

extension ViewController {
    @objc func alamofirebtnAction() {
        self.present(AlamofireTestVC(), animated: true, completion: nil)
    }
}


